Sources/Internal/LayoutItemTypeEnumerator.swift

```diff
@@ -127,13 +132,16 @@ final class LayoutItemTypeEnumerator {
     case .day(let day):
       let nextDay = calendar.day(byAddingDays: 1, to: day)
       if day.month != nextDay.month {
-        return .monthHeader(nextDay.month)
+        return .monthFooter(day.month)
       } else if day == dayRange.upperBound {
         let nextMonth = calendar.month(byAddingMonths: 1, to: nextDay.month)
         return .monthHeader(nextMonth)
       } else {
         return .day(nextDay)
       }
+    
+    case .monthFooter(let month):
+      return .monthHeader(calendar.month(byAddingMonths: 1, to: month))
```

Comment:
```suggestion
      let nextMonth = calendar.month(byAddingMonths: 1, to: month)
      return .monthHeader(nextMonth)
```

For consistency with the rest of the cases ^